lighthouse-core/fraggle-rock/config/filters.js

```diff
@@ -36,20 +36,24 @@ function filterArtifactsByGatherMode(artifacts, mode) {
 
 /**
  * Filters an array of audits down to the set that can be computed using only the specified artifacts.
+ * Optional `applicableModes` property can explicitly exclude an audit even if all required artifacts are available.
  *
  * @param {LH.Config.FRConfig['audits']} audits
  * @param {Array<LH.Config.AnyArtifactDefn>} availableArtifacts
+ * @param {LH.Gatherer.GatherMode} mode
  * @return {LH.Config.FRConfig['audits']}
  */
-function filterAuditsByAvailableArtifacts(audits, availableArtifacts) {
+function filterAudits(audits, availableArtifacts, mode) {
```

Comment:
separate function `filterAuditsByGatherMode`? the `BySomeVerboseSingleThing` was intentional for clarity :)

My lessons from the way config evolved the past 5 years and how important these files are was to be as excessively clear as possible in v2 ;)

Reply:
I tried `filterAuditsByGatherModeAndAvailableArtifacts` but that was too long. I'm ok excluding one of the filters from the name if you are though.

Reply:
Now that I think about it `filterAuditsByAvailableArtifacts` makes more sense because we will filtering by artifacts most of the time.

Reply:
WDYT about two functions?

`filterAuditsByAvailableArtifacts`
`filterAuditsByGatherMode`

`filterCategories` invokes both of them in sequence